src/pages/authentication/Login.jsx

```diff
@@ -0,0 +1,149 @@
+      { ...loginFormState },
+      false
+    );
+    const { data, success } = response;
+    if (success) {
+      authDispatch({ type: "SET_LOGGED_USER", payload: data });
+      location.state === null
+        ? navigate("/scroll")
+        : navigate(location?.state?.from);
+    } else {
+      alert("Something went wrong, check console");
+    }
+  };
+
+  const loginUser = (e) => {
+    e.preventDefault();
+    const errors = validateForm(loginFormState);
+    if (Object.keys(errors).length === 0) {
+      setFormError(inititalLoginState);
+      loginUserApiCall();
+    } else {
+      setFormError(errors);
+    }
+  };
+  const inputChangeHandler = (e) => {
+    setFormError(inititalLoginState);
+    const new_val = { [e.target.name]: e.target.value };
+    setLoginFormState((prev) => ({ ...prev, ...new_val }));
+  };
+  const fillDemoCredentials = () => {
+    setLoginFormState((prev) => ({
+      ...prev,
+      username: "adarshbalika",
+      password: "adarshBalika123",
+    }));
+  };
+  return (
+    <div className="box-border w-full p-1">
+      <div className="w-full md:max-w-md mx-auto border-2 md:border-r-4 border-black rounded-md flex flex-col gap-5 py-10">
+        <p className="text-3xl font-semibold w-max mx-auto ">
+          Log in to Peerlist
+        </p>
+        <form className="bg-white  rounded px-8  mb-4" onSubmit={loginUser}>
+          <div className="mb-4">
+            <label
+              className="block text-gray-700 text-sm mb-2 border border-gray-400 rounded px-2 focus-within:border-black"
+              htmlFor="username"
+            >
+              Username
+              <input
+                className="appearance-none w-full my-2 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
+                id="username"
+                name="username"
+                type="text"
+                value={loginFormState.username}
+                onChange={inputChangeHandler}
+              />
+            </label>
+            <p className="text-xs text-red-500">{formError.username}</p>
+          </div>
+          <div className="mb-6">
+            <label
+              className="block text-gray-700 text-sm mb-2 border border-gray-400 rounded px-2 focus-within:border-black"
+              htmlFor="password"
+            >
+              Password
+              <div className="relative">
+                <input
+                  className="appearance-none w-full text-gray-700 mb-3 leading-tight focus:outline-none focus:shadow-outline"
+                  id="password"
+                  name="password"
+                  type={showPassword ? "text" : "password"}
+                  value={loginFormState.password}
+                  onChange={inputChangeHandler}
+                />
+                <button
+                  className="flex absolute inset-y-0 right-0 bg-hover-color h-6 px-1 rounded-md"
+                  onClick={() => setShowPassword((prev) => !prev)}
```

Comment:
pass `event` and add `event.preventDefault()` to prevent form submit

Reply:
thanks for this i was not able to figure why form is validating on show/hide.